core/src/main/java/org/mule/runtime/core/api/processor/strategy/ProcessingStrategy.java

```diff
@@ -48,7 +48,14 @@ default void registerInternalSink(Publisher<CoreEvent> flux, String sinkRepresen
     Flux.from(flux).subscribe();
   }
 
-  default Publisher<CoreEvent> registerInternalFlux(Publisher<CoreEvent> flux) {
+  /**
+   * For fluxes created outside of the main flux for a flow, have them accounted for in the processing strategy for a graceful
```

Comment:
is graceful shutdown a particular use case of this method or is the only and sole purpose of it? Depending on that, either change the jdoc or rename to something more specific